KanBanApi/KanBanApi.Application/Commands/GoogleSignIn/GoogleSignInCommandHandler.cs

```diff
@@ -0,0 +1,39 @@
+using FastEndpoints;
+using Google.Apis.Auth;
+using KanBanApi.Application.Dtos;
+using KanBanApi.Application.Services;
+using Microsoft.AspNetCore.Identity;
+
+namespace KanBanApi.Application.Commands.GoogleSignIn;
+
+public class GoogleSignInCommandHandler(UserManager<IdentityUser> _userManager, IJwtGenerator _jwtGenerator, GoogleConfigDto _googleConfig) : CommandHandler<GoogleSignInCommand, UserResponse>
+{
+    public override async Task<UserResponse> ExecuteAsync(GoogleSignInCommand command, CancellationToken ct = default)
+    {
+        var validationSettings = new GoogleJsonWebSignature.ValidationSettings { Audience = new[] { _googleConfig.ClientId } };
+        var payload = await GoogleJsonWebSignature.ValidateAsync(command.GoogleToken, validationSettings);
+
+        var user = await _userManager.FindByEmailAsync(payload.Email);
+
+        if (user == null)
+        {
+            user = new IdentityUser
+            {
+                Email = payload.Email,
+                UserName = GetUserName(payload.Email)
+            };
+
+            await _userManager.CreateAsync(user);
+        }
+
+
```

Comment:
Same applies here I believe. Although the simplificy of the code has a value here, and is a good thing, the separation of concerns and the single responsibility principle is something we should observe in real life projects.